seaworthy/tests-pytest/test_fixtures.py

```diff
@@ -81,20 +89,97 @@ def test_setup_teardown(self, docker_helper):
             next(fixture_gen)
 
         # Container has been stopped and removed
-        with pytest.raises(RuntimeError):
-            container.inner()
+        assert not container.created
+
+    def test_dependencies(self, request, docker_helper):
+        """
+        When the fixture depends on other fixtures, those fixtures should be
+        setup when the fixture is used.
+        """
+        container_fixture = resource_fixture(
+            ContainerDefinition(name='test', image=IMG), 'container_test',
+            dependencies=('volume_test', 'network_test'))
+        fixture_gen = container_fixture(request, docker_helper)
+        container = next(fixture_gen)
+
+        assert isinstance(container, ContainerDefinition)
+        assert container.status() == 'running'
+        assert volume.created
+        assert network.created
+
+        # Test things are torn down
+        with pytest.raises(StopIteration):
+            next(fixture_gen)
+
+        # Container has been stopped and removed
+        assert not container.created
+        # Can't assert dependent fixture teardown--only happens once test
+        # method returns
 
 
 @dockertest()
 class TestCleanContainerFixturesFunc:
-    def test_setup_teardown(self, docker_helper):
+    def test_setup_teardown(self, request, docker_helper):
         """
         The fixture should yield a started container, and afterwards stop and
         remove the container.
         """
         raw_fixture, fixture = clean_container_fixtures(
             ContainerDefinition(name='test', image=IMG), 'test')
-        fixture_gen = raw_fixture(docker_helper)
+        fixture_gen = raw_fixture(request, docker_helper)
+        # TODO: Assert on cleaning fixture
```

Comment:
Is this TODO going to be addressed in this PR?

Reply:
Wasn't planning to :( it's carried over from a similar test.